File: android/app/src/main/kotlin/com/saudepilates/app/util/ToastManager.kt
package com.saudepilates.app.util

import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow

enum class ToastType { SUCCESS, ERROR, WARNING, INFO }

data class ToastMessage(val message: String, val type: ToastType = ToastType.SUCCESS)

object ToastManager {
    private val _toast = MutableStateFlow<ToastMessage?>(null)
    val toast: StateFlow<ToastMessage?> = _toast.asStateFlow()

    fun success(message: String) { show(message, ToastType.SUCCESS) }
    fun error(message: String) { show(message, ToastType.ERROR) }
    fun warning(message: String) { show(message, ToastType.WARNING) }
    fun info(message: String) { show(message, ToastType.INFO) }
    fun clear() { _toast.value = null }

    private fun show(message: String, type: ToastType) {
        _toast.value = ToastMessage(message, type)
    }
}

File: ios/SaudePilates/Utilities/ToastManager.swift
import SwiftUI

enum ToastType {
    case success, error, warning, info
}

struct ToastMessage: Identifiable, Equatable {
    let id = UUID()
    let message: String
    let type: ToastType
}

@MainActor
final class ToastManager: ObservableObject {
    static let shared = ToastManager()

    @Published var currentToast: ToastMessage?

    func show(_ message: String, type: ToastType = .success) {
        currentToast = ToastMessage(message: message, type: type)
        Task {
            try? await Task.sleep(nanoseconds: 4_000_000_000)
            if currentToast?.message == message {
                currentToast = nil
            }
        }
    }

    func success(_ message: String) { show(message, type: .success) }
    func error(_ message: String) { show(message, type: .error) }
}
